GccBoardProject1/src/peopleCounting.c

```diff
@@ -8,139 +8,164 @@
 #include <asf.h>
 #include <grideye.h>
 #include <peopleCounting.h>
+#include <numpyInC.h>
 
 
-bool is_local_max(uint16_t ** frame, uint16_t row, uint16_t col) {
-	uint16_t current_max = frame[row][col];
+bool is_local_max(uint16_t COUNT_BUFF[], uint16_t frame_index, uint16_t row, uint16_t col) {
+	convertToFrame(COUNT_BUFF, frame_index);
+	uint16_t current_max = frame_conv[row][col];
+	
 	// Greater than (row+1, col), (row-1, col)
-	if (current_max < frame[row+1][col] || current_max < frame[row-1][col]) {
+	if (current_max < frame_conv[row+1][col] || current_max < frame_conv[row-1][col]) {
 		return false;
 	}
 	// Greater than (row, col+1), (row, col-1)
-	if (current_max < frame[row][col+1] || current_max < frame[row][col-1]) {
+	if (current_max < frame_conv[row][col+1] || current_max < frame_conv[row][col-1]) {
 		return false;
 	}
 	// Greater than (row+1, col+1), (row-1, col-1)
-	if (current_max < frame[row+1][col+1] || current_max < frame[row-1][col-1]) {
+	if (current_max < frame_conv[row+1][col+1] || current_max < frame_conv[row-1][col-1]) {
 		return false;
 	}
 	// Greater than (row+1, col-1), (row-1, col+1)
-	if (current_max < frame[row+1][col-1] || current_max < frame[row-1][col+1]) {
+	if (current_max < frame_conv[row+1][col-1] || current_max < frame_conv[row-1][col+1]) {
 		return false;
 	}
 	return true;
 }
 
-uint16_t *** COUNT_BUFF addFrame(uint16_t ** pix_buff, uint16_t*** COUNT_BUFF, uint16_t frameCount) {
+uint16_t* addFrame(uint16_t PIXEL_BUFFER[], uint16_t COUNT_BUFF[], uint16_t frameCount, struct keepCount *count) {
 	if (frameCount < 7) {
-		COUNT_BUFF[frameCount-1] = pix_buff;
+		for (int i = 0; i < NUM_PIXELS; i++) {
+			COUNT_BUFF[(frameCount-1)*NUM_PIXELS+i] = PIXEL_BUFFER[i];
+		}
 	}
 	else {
-		for (int i = 0; i < 6; i++){
-			COUNT_BUFF[i] = COUNT_BUFF[i+1];
+		for (int i = 0; i < 6*NUM_PIXELS; i++){
+			COUNT_BUFF[i] = COUNT_BUFF[NUM_PIXELS+i];
+		}
+		for (int i = 0; i < NUM_PIXELS; i++) {
+			COUNT_BUFF[7*NUM_PIXELS+i] = PIXEL_BUFFER[i];
 		}
-		COUNT_BUFF[7] = pix_buff;
 	}
+	count->counted = false;
 	return COUNT_BUFF;
 }
 
-uint16_t getMaxId(uint16_t ** array, uint16_t col) {
-	uint16_t maxNum = array[0];
+uint16_t getMaxId(uint16_t COUNT_BUFF[], uint16_t ind, uint16_t col) {
+	convertToFrame(COUNT_BUFF, ind);
+	
+	uint16_t maxNum = frame_conv[0][0];
 	uint16_t maxRowId = 0;
 	
 	for (int i=1; i < GRID_SIZE; i++){
-		if (maxNum > array[i][col]) {
-			maxNum = array[i][col];
+		if (maxNum > frame_conv[i][col]) {
+			maxNum = frame_conv[i][col];
 			maxRowId = i;
 		}
 	}
 	return maxRowId;
 }
 
+void convertToFrame(uint16_t COUNT_BUFF[], uint16_t ind) {
+	for (int i = 0; i < GRID_SIZE; i++) {
+		for (int j = 0; j < GRID_SIZE; j++) {
+			frame_conv[i][j] = COUNT_BUFF[(ind*NUM_PIXELS) + GRID_SIZE*i + j];
```

Comment:
After reading this code more, I think the cleaner way to deal with stacks of 2D frames is to leave them all as 1D. That way, we don't have to distinguish between `uint16_t xxx[][]` and `uint16_t **xxx`. The backbone of the frame stack can have the same structure that you have (where it holds pointers) and for all the functions that are easier to write as `[row][col]` we can write:
 - `uint16_t compute_index(uint16_t row, uint16_t col)` so that we can do `frame[compute_index(r,c)]`
 - Alternatively, `uint16_t get_frame_value(uint16_t *frame, uint16_t row, uint16_t col)`
 - Also, `uint16_t get_value_at_time(unint16_t **frame_stack, uint16_t index, uint16_t row, uint16_t col)`